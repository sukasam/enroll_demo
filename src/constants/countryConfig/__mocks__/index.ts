export const useCountryConfig = jest.fn().mockReturnValue({
    marketExceptions: {
        showOnboardingGuide: true,
        hideUnicityPhone: false,
        thankYouImage: "/img/custom-image.jpg"
    }
});

export const getSuppotedCountries = jest.fn();
export const getUnsupportedCountries = jest.fn();
export const getCountries = jest.fn();
export const getCountryConfig = jest.fn();
export const isSupported = jest.fn();
export const getDefaultLanguage = jest.fn();
export const countryConfigs = [];
