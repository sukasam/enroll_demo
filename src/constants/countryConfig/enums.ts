export enum Alpha3 {
    USA = "USA",
    PRI = "PRI",
    CAN = "CAN",
    GBR = "GBR",
    ARE = "ARE",
    AIA = "AIA",
    AUT = "AUT",
    AUS = "AUS",
    BRB = "BRB",
    BEL = "BEL",
    BHR = "BHR",
    BES = "BES",
    BHS = "BHS",
    CHE = "CHE",
    COL = "COL",
    CUW = "CUW",
    CZE = "CZE",
    DEU = "DEU",
    DNK = "DNK",
    DMA = "DMA",
    DOM = "DOM",
    DZA = "DZA",
    EGY = "EGY",
    ESP = "ESP",
    FRA = "FRA",
    GRD = "GRD",
    HUN = "HUN",
    IRL = "IRL",
    IND = "IND",
    ITA = "ITA",
    JOR = "JOR",
    JPN = "JPN",
    KEN = "KEN",
    KOR = "KOR",
    KWT = "KWT",
    LBN = "LBN",
    LUX = "LUX",
    MEX = "MEX",
    NLD = "NLD",
    NOR = "NOR",
    NZL = "NZL",
    OMN = "OMN",
    PHL = "PHL",
    POL = "POL",
    SAU = "SAU",
    SWE = "SWE",
    SGP = "SGP",
    TCA = "TCA",
    THA = "THA",
    TUR = "TUR",
    TTO = "TTO",
    UKR = "UKR",
    VIR = "VIR",
    MAR = "MAR",
    LAO = "LAO",
    MYS = "MYS",
    MMR = "MMR",
    TWN = "TWN",
    VNM = "VNM",
    HKG = "HKG",
    KHM = "KHM",
    IDN = "IDN",
    JAM = "JAM",
    AFG = "AFG",
    ALA = "ALA",
    ALB = "ALB",
    ASM = "ASM",
    AND = "AND",
    AGO = "AGO",
    ATA = "ATA",
    ATG = "ATG",
    ARG = "ARG",
    ARM = "ARM",
    ABW = "ABW",
    AZE = "AZE",
    BGD = "BGD",
    BLR = "BLR",
    BLZ = "BLZ",
    BEN = "BEN",
    BMU = "BMU",
    BTN = "BTN",
    BOL = "BOL",
    BIH = "BIH",
    BWA = "BWA",
    BVT = "BVT",
    BRA = "BRA",
    IOT = "IOT",
    BRN = "BRN",
    BGR = "BGR",
    BFA = "BFA",
    BDI = "BDI",
    CMR = "CMR",
    CPV = "CPV",
    CYM = "CYM",
    CAF = "CAF",
    TCD = "TCD",
    CHL = "CHL",
    CHN = "CHN",
    CXR = "CXR",
    CCK = "CCK",
    COM = "COM",
    COK = "COK",
    CRI = "CRI",
    CIV = "CIV",
    HRV = "HRV",
    CUB = "CUB",
    CYP = "CYP",
    COD = "COD",
    DJI = "DJI",
    ECU = "ECU",
    SLV = "SLV",
    GNQ = "GNQ",
    ERI = "ERI",
    EST = "EST",
    ETH = "ETH",
    FLK = "FLK",
    FRO = "FRO",
    FJI = "FJI",
    FIN = "FIN",
    GUF = "GUF",
    PYF = "PYF",
    ATF = "ATF",
    GAB = "GAB",
    GEO = "GEO",
    GHA = "GHA",
    GIB = "GIB",
    GRC = "GRC",
    GRL = "GRL",
    GLP = "GLP",
    GUM = "GUM",
    GTM = "GTM",
    GGY = "GGY",
    GIN = "GIN",
    GNB = "GNB",
    GCC = "GCC",
    GUY = "GUY",
    HTI = "HTI",
    HMD = "HMD",
    HND = "HND",
    ISL = "ISL",
    IRN = "IRN",
    IRQ = "IRQ",
    IMN = "IMN",
    ISR = "ISR",
    JEY = "JEY",
    KAZ = "KAZ",
    KIR = "KIR",
    KGZ = "KGZ",
    LVA = "LVA",
    LSO = "LSO",
    LBR = "LBR",
    LBY = "LBY",
    LIE = "LIE",
    LTU = "LTU",
    MAC = "MAC",
    MDG = "MDG",
    MWI = "MWI",
    MDV = "MDV",
    MLI = "MLI",
    MLT = "MLT",
    MHL = "MHL",
    MTQ = "MTQ",
    MRT = "MRT",
    MUS = "MUS",
    MYT = "MYT",
    FSM = "FSM",
    MDA = "MDA",
    MCO = "MCO",
    MNG = "MNG",
    MNE = "MNE",
    MSR = "MSR",
    MOZ = "MOZ",
    NAM = "NAM",
    NRU = "NRU",
    NPL = "NPL",
    NCL = "NCL",
    NIC = "NIC",
    NER = "NER",
    NGA = "NGA",
    NIU = "NIU",
    NFK = "NFK",
    PRK = "PRK",
    MNP = "MNP",
    PAK = "PAK",
    PLW = "PLW",
    PAN = "PAN",
    PNG = "PNG",
    PRY = "PRY",
    PER = "PER",
    PCN = "PCN",
    PRT = "PRT",
    QAT = "QAT",
    MKD = "MKD",
    COG = "COG",
    REU = "REU",
    ROU = "ROU",
    RUS = "RUS",
    RWA = "RWA",
    BLM = "BLM",
    SHN = "SHN",
    KNA = "KNA",
    LCA = "LCA",
    MAF = "MAF",
    SPM = "SPM",
    VCT = "VCT",
    WSM = "WSM",
    SMR = "SMR",
    STP = "STP",
    SEN = "SEN",
    SRB = "SRB",
    SYC = "SYC",
    SLE = "SLE",
    SXM = "SXM",
    SVK = "SVK",
    SVN = "SVN",
    SLB = "SLB",
    SOM = "SOM",
    ZAF = "ZAF",
    SGS = "SGS",
    SSD = "SSD",
    LKA = "LKA",
    PSE = "PSE",
    SDN = "SDN",
    SUR = "SUR",
    SJM = "SJM",
    SWZ = "SWZ",
    SYR = "SYR",
    TJK = "TJK",
    TZA = "TZA",
    GMB = "GMB",
    TLS = "TLS",
    TGO = "TGO",
    TKL = "TKL",
    TON = "TON",
    TUN = "TUN",
    TKM = "TKM",
    TUV = "TUV",
    UGA = "UGA",
    UMI = "UMI",
    URY = "URY",
    UZB = "UZB",
    VUT = "VUT",
    VAT = "VAT",
    VEN = "VEN",
    VGB = "VGB",
    WLF = "WLF",
    ESH = "ESH",
    YEM = "YEM",
    ZMB = "ZMB",
    ZWE = "ZWE"
}

export enum Alpha2 {
    US = "US",
    PR = "PR",
    CA = "CA",
    GB = "GB",
    AE = "AE",
    AI = "AI",
    AT = "AT",
    AU = "AU",
    BB = "BB",
    BE = "BE",
    BH = "BH",
    BQ = "BQ",
    BS = "BS",
    CH = "CH",
    CO = "CO",
    CW = "CW",
    CZ = "CZ",
    DE = "DE",
    DK = "DK",
    DM = "DM",
    DO = "DO",
    DZ = "DZ",
    EG = "EG",
    ES = "ES",
    FR = "FR",
    GD = "GD",
    HU = "HU",
    IE = "IE",
    IN = "IN",
    IT = "IT",
    JO = "JO",
    JP = "JP",
    KE = "KE",
    KR = "KR",
    KW = "KW",
    LB = "LB",
    LU = "LU",
    MX = "MX",
    NL = "NL",
    NO = "NO",
    NZ = "NZ",
    OM = "OM",
    PH = "PH",
    PL = "PL",
    SA = "SA",
    SE = "SE",
    SG = "SG",
    TC = "TC",
    TH = "TH",
    TR = "TR",
    TT = "TT",
    UA = "UA",
    VI = "VI",
    MA = "MA",
    LA = "LA",
    MY = "MY",
    MM = "MM",
    TW = "TW",
    VN = "VN",
    HK = "HK",
    KH = "KH",
    ID = "ID",
    JM = "JM",
    AF = "AF",
    AX = "AX",
    AL = "AL",
    AS = "AS",
    AD = "AD",
    AO = "AO",
    AQ = "AQ",
    AG = "AG",
    AR = "AR",
    AM = "AM",
    AW = "AW",
    AZ = "AZ",
    BD = "BD",
    BY = "BY",
    BZ = "BZ",
    BJ = "BJ",
    BM = "BM",
    BT = "BT",
    BO = "BO",
    BA = "BA",
    BW = "BW",
    BV = "BV",
    BR = "BR",
    IO = "IO",
    BN = "BN",
    BG = "BG",
    BF = "BF",
    BI = "BI",
    CM = "CM",
    CV = "CV",
    KY = "KY",
    CF = "CF",
    TD = "TD",
    CL = "CL",
    CN = "CN",
    CX = "CX",
    CC = "CC",
    KM = "KM",
    CK = "CK",
    CR = "CR",
    CI = "CI",
    HR = "HR",
    CU = "CU",
    CY = "CY",
    CD = "CD",
    DJ = "DJ",
    EC = "EC",
    SV = "SV",
    GQ = "GQ",
    ER = "ER",
    EE = "EE",
    ET = "ET",
    FK = "FK",
    FO = "FO",
    FJ = "FJ",
    FI = "FI",
    GF = "GF",
    PF = "PF",
    TF = "TF",
    GA = "GA",
    GE = "GE",
    GH = "GH",
    GI = "GI",
    GR = "GR",
    GL = "GL",
    GP = "GP",
    GU = "GU",
    GT = "GT",
    GG = "GG",
    GN = "GN",
    GW = "GW",
    GC = "GC",
    GY = "GY",
    HT = "HT",
    HM = "HM",
    HN = "HN",
    IS = "IS",
    IR = "IR",
    IQ = "IQ",
    IM = "IM",
    IL = "IL",
    JE = "JE",
    KZ = "KZ",
    KI = "KI",
    KG = "KG",
    LV = "LV",
    LS = "LS",
    LR = "LR",
    LY = "LY",
    LI = "LI",
    LT = "LT",
    MO = "MO",
    MG = "MG",
    MW = "MW",
    MV = "MV",
    ML = "ML",
    MT = "MT",
    MH = "MH",
    MQ = "MQ",
    MR = "MR",
    MU = "MU",
    YT = "YT",
    FM = "FM",
    MD = "MD",
    MC = "MC",
    MN = "MN",
    ME = "ME",
    MS = "MS",
    MZ = "MZ",
    NA = "NA",
    NR = "NR",
    NP = "NP",
    NC = "NC",
    NI = "NI",
    NE = "NE",
    NG = "NG",
    NU = "NU",
    NF = "NF",
    KP = "KP",
    MP = "MP",
    PK = "PK",
    PW = "PW",
    PA = "PA",
    PG = "PG",
    PY = "PY",
    PE = "PE",
    PN = "PN",
    PT = "PT",
    QA = "QA",
    MK = "MK",
    CG = "CG",
    RE = "RE",
    RO = "RO",
    RU = "RU",
    RW = "RW",
    BL = "BL",
    SH = "SH",
    KN = "KN",
    LC = "LC",
    MF = "MF",
    PM = "PM",
    VC = "VC",
    WS = "WS",
    SM = "SM",
    ST = "ST",
    SN = "SN",
    RS = "RS",
    SC = "SC",
    SL = "SL",
    SX = "SX",
    SK = "SK",
    SI = "SI",
    SB = "SB",
    SO = "SO",
    ZA = "ZA",
    GS = "GS",
    SS = "SS",
    LK = "LK",
    PS = "PS",
    SD = "SD",
    SR = "SR",
    SJ = "SJ",
    SZ = "SZ",
    SY = "SY",
    TJ = "TJ",
    TZ = "TZ",
    GM = "GM",
    TL = "TL",
    TG = "TG",
    TK = "TK",
    TO = "TO",
    TN = "TN",
    TM = "TM",
    TV = "TV",
    UG = "UG",
    UM = "UM",
    UY = "UY",
    UZ = "UZ",
    VU = "VU",
    VA = "VA",
    VE = "VE",
    VG = "VG",
    WF = "WF",
    EH = "EH",
    YE = "YE",
    ZM = "ZM",
    ZW = "ZW"
}

export enum Currency {
    USD = "USD",
    CAD = "CAD",
    MXN = "MXN",
    GBP = "GBP",
    EUR = "EUR",
    CHF = "CHF",
    COP = "COP",
    JPY = "JPY",
    PHP = "PHP"
}

export enum optInNotificationTypes {
    EMAIL = "EMAIL",
    SMS = "SMS"
}
