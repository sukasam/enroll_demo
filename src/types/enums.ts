// eslint-disable-next-line import/prefer-default-export
export enum CustomerType {
    MEMBER = "Member",
    ASSOCIATE = "Associate",
    CUSTOMER = "Customer"
}

export enum PaymentType {
    CreditCard = "credit_card",
    CreditCardAlias = "credit_card_alias",
    PayPal = "paypal",
    PayPal3ds = "paypal_3ds",
    Cash = "cash",
    BankWire = "bankwire",
    BankDraft = "bank_draft",
    EWalletToken = "ewallet_token",
    SavedCreditCard = "saved_credit_card"
}

export enum MixpanelPaymentMethod {
    CreditCard = "credit_card",
    CreditCardWorldPay = "credit_card_worldpay",
    PayPal3ds = "paypal_3ds",
    PayPal = "paypal",
    Cash = "cash",
    BankWire = "bankwire"
}
