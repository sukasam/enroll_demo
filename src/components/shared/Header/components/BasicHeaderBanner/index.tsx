import BannerNotificationAlerts from "Components/shared/BannerNotificationAlerts/index";
import { Alpha2 } from "Constants/countryConfig/enums";
import { useFlags } from "launchdarkly-react-client-sdk";
import { useRouter } from "next/router";

interface BasicHeaderBannerProps {
    currentMarket: Alpha2;
    flagNameLD: string;
}

export default function BasicHeaderBanner(
    props: BasicHeaderBannerProps
): JSX.Element | null {
    const { currentMarket, flagNameLD } = props;
    const flags = useFlags();
    const flagValue = flags[flagNameLD]; // Access the flag by its string name
    const router = useRouter();

    if (router.pathname.includes("/home") && flagNameLD === "outOfStock") {
        return null; // Do not render the banner on the Home page
    }

    return (
        <BannerNotificationAlerts
            currentMarket={currentMarket}
            rosettaAlterations={(displayMessage: string): string =>
                displayMessage
            }
            bannerFlagLD={flagValue}
        />
    );
}
