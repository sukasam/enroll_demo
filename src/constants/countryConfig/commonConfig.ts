import { Currency, optInNotificationTypes } from "./enums";
import {
    MarketExceptions,
    PaymentLogo,
    PaymentOptions,
    ProductConfig,
    TemplateSupportedCountryConfig
} from "./types";

const productConfig: ProductConfig = {
    recommendedPack: 0,
    packs: [
        {
            isDigital: false,
            buttonName: "catalog_get_business_builder",
            kitDescription: "business_builder_kit_description",
            mainFeatures: [
                "catalog_modal_business_builder_main_features_1",
                "catalog_modal_business_builder_main_features_2",
                "catalog_modal_business_builder_main_features_3",
                "catalog_modal_business_builder_main_features_4",
                "catalog_modal_business_builder_main_features_5",
                "catalog_modal_business_builder_main_features_6",
                "catalog_modal_business_builder_main_features_7",
                "catalog_modal_business_builder_main_features_8"
            ],
            accentColor: "#FAA977",
            backgroundColor: "#FBF7F1",
            perks: [
                {
                    icon: "/svg/page-with-checkmark.svg",
                    perk: "business_builder_perk_1"
                },
                {
                    icon: "/svg/phone-with-checkmark.svg",
                    perk: "business_builder_perk_2"
                },
                {
                    icon: "/svg/add-people.svg",
                    perk: "business_builder_perk_3"
                }
            ],
            extras: [
                {
                    icon: "/svg/dollar-checkmark-circle.svg",
                    title: "business_builder_extra_1_title",
                    description: "business_builder_extra_1_description"
                }
            ],
            products: [
                {
                    image: "/img/unimate-lemon-3.png",
                    title: "business_builder_product_1_title",
                    description: "business_builder_product_1_description"
                },
                {
                    image: "/img/balance-3.png",
                    title: "business_builder_product_2_title",
                    description: "business_builder_product_2_description"
                }
            ],
            annotations: "business_builder_annotations",
            showPaymentType: true
        },
        {
            name: "basic_name",
            buttonName: "catalog_basic_distributor_only",
            description: "basic_description",
            isDigital: true,
            showPaymentType: true
        }
    ]
};

const paymentOptionsConfig: PaymentOptions = {
    hasPayPal: false,
    hasCash: false,
    hasBankWire: false,
    hasCreditCard: false,
    hasSavedCreditCard: false
};

const marketExceptions: MarketExceptions = {
    zipLength: [5, 9],
    taxIdPattern: undefined,
    hasSecondaryTaxId: false,
    taxInclusiveBackEnd: false,
    taxInclusiveFrontEnd: false,
    hiddenShippingProvince: true,
    useQuotesToValidateAddress: true,
    hasCommissionPaymentOptions: true,
    showOnboardingGuide: true,
    showVisaAgreement: false,
    paymentOptions: paymentOptionsConfig,
    useWorldPay: false,
    usePayPal3ds: false,
    notificationType: optInNotificationTypes.SMS,
    hiddenDecimal: false,
    updateMainAddress: true,
    formCreateAccount: [
        "firstName",
        "lastName",
        "phoneNumber",
        "email",
        "password"
    ],
    formUpdateAccount: ["firstName", "lastName", "phoneNumber", "email"]
};

const commonConfig: TemplateSupportedCountryConfig = {
    dateFormat: "MDY",
    languages: [
        {
            code: "en",
            name: "English",
            default: true
        }
    ],
    currency: Currency.USD,
    alwaysShowGuide: false,
    paymentLogos: [
        PaymentLogo.Visa,
        PaymentLogo.Mastercard,
        PaymentLogo.Amex,
        PaymentLogo.Discover,
        PaymentLogo.JCB
    ],
    productConfig,
    marketExceptions
};

export default commonConfig;
