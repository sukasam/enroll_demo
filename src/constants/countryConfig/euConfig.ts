import { Currency } from "./enums";
import {
    MarketExceptions,
    PaymentLogo,
    PaymentOptions,
    ProductConfig,
    TemplateSupportedCountryConfig
} from "./types";

const productConfig: ProductConfig = {
    recommendedPack: 0,
    packs: [
        {
            isDigital: false,
            buttonName: "catalog_get_business_builder",
            kitDescription: "business_builder_kit_description",
            mainFeatures: [
                "catalog_modal_business_builder_main_features_1",
                "catalog_modal_business_builder_main_features_2",
                "catalog_modal_business_builder_main_features_3",
                "catalog_modal_business_builder_main_features_4",
                "catalog_modal_business_builder_main_features_5",
                "catalog_modal_business_builder_main_features_6",
                "catalog_modal_business_builder_main_features_7",
                "catalog_modal_business_builder_main_features_8"
            ],
            accentColor: "#FAA977",
            backgroundColor: "#FBF7F1",
            perks: [
                {
                    icon: "/svg/page-with-checkmark.svg",
                    perk: "business_builder_perk_1"
                },
                {
                    icon: "/svg/phone-with-checkmark.svg",
                    perk: "business_builder_perk_2"
                },
                {
                    icon: "/svg/add-people.svg",
                    perk: "business_builder_perk_3"
                }
            ],
            extras: [
                {
                    icon: "/svg/dollar-checkmark-circle.svg",
                    title: "business_builder_extra_1_title",
                    description: "business_builder_extra_1_description"
                }
            ],
            products: [
                {
                    image: "/img/unimate-lemon-3.png",
                    title: "business_builder_product_1_title",
                    description: "business_builder_product_1_description"
                },
                {
                    image: "/img/balance-3.png",
                    title: "business_builder_product_2_title",
                    description: "business_builder_product_2_description"
                }
            ],
            annotations: "business_builder_annotations",
            showPaymentType: true
        },
        {
            name: "basic_name",
            buttonName: "catalog_basic_distributor_only",
            description: "basic_description",
            isDigital: true,
            showPaymentType: true
        }
    ]
};

const paymentOptionsConfig: PaymentOptions = {
    hasPayPal: true,
    hasCash: false,
    hasBankWire: true,
    hasCreditCard: true,
    hasSavedCreditCard: false,
    hasKlarna: false
};

const marketExceptions: MarketExceptions = {
    zipLength: 4,
    taxIdPattern: undefined,
    disableTaxId: true,
    taxInclusiveBackEnd: false,
    taxInclusiveFrontEnd: false,
    hiddenShippingProvince: true,
    useQuotesToValidateAddress: true,
    hasCommissionPaymentOptions: true,
    showOnboardingGuide: false,
    showVisaAgreement: false,
    paymentOptions: paymentOptionsConfig,
    useWorldPay: true,
    usePayPal3ds: true,
    hiddenDecimal: false,
    useGlobalFormatCurrency: true,
    hiddenPersonalData: true,
    useCookie: true,
    updateMainAddress: true
};

const euConfig: TemplateSupportedCountryConfig = {
    dateFormat: "DMY",
    languages: [
        {
            code: "en",
            name: "English",
            default: true
        }
    ],
    currency: Currency.EUR,
    alwaysShowGuide: true,
    paymentLogos: [PaymentLogo.Visa, PaymentLogo.Mastercard],
    productConfig,
    marketExceptions
};

export default euConfig;
