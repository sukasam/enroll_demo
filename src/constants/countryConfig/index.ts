import { useTranslations } from "Contexts/translation";
import { useMemo } from "react";
import countries from "./countries";
import { Alpha2 } from "./enums";
import {
    CountryConfig,
    Locale,
    SupportedCountryConfig,
    UnsupportedCountryConfig
} from "./types";

// This makes importing less confusing
export { countries as countryConfigs };

export function getSuppotedCountries(): SupportedCountryConfig[] {
    return countries.filter<SupportedCountryConfig>(
        (config): config is SupportedCountryConfig => config.isSupported
    );
}

export function getUnsupportedCountries(): UnsupportedCountryConfig[] {
    return countries.filter(
        (config): config is UnsupportedCountryConfig => !config.isSupported
    );
}

export function getCountries(supported: boolean): CountryConfig[] {
    if (supported) {
        return getSuppotedCountries();
    }

    return countries;
}

export function getCountryConfig(
    countryCode: Alpha2,
    supportedOnly: true
): SupportedCountryConfig | undefined;

export function getCountryConfig(
    countryCode: Alpha2,
    supportedOnly: false
): CountryConfig | undefined;

export function getCountryConfig(
    countryCode: Alpha2,
    supportedOnly: boolean
): CountryConfig | undefined {
    if (supportedOnly) {
        return getSuppotedCountries().find(
            config => config.alpha2 === countryCode
        );
    }

    return countries.find(config => config.alpha2 === countryCode);
}

export function isSupported(
    config: CountryConfig
): config is SupportedCountryConfig {
    return config.isSupported;
}

export function useCountryConfig(): SupportedCountryConfig | undefined {
    const { country } = useTranslations();
    const alpha2 = country as Alpha2;
    const config = useMemo(() => getCountryConfig(alpha2, true), [alpha2]);
    return config;
}

export function getDefaultLanguage(alpha2: Alpha2): string {
    const config = getCountryConfig(alpha2, true);

    return config?.languages.find(lang => lang.default)?.code ?? "en";
}

export function getValidatedLocale(
    countryCode: string | null | Alpha2,
    language: string | null
): Locale {
    // Default to US if no country or invalid country
    const countryConfig =
        getCountryConfig(countryCode as Alpha2, true) ||
        getCountryConfig(Alpha2.US, true);

    // Get default language for the country
    const defaultLanguage = getDefaultLanguage(countryCode as Alpha2);

    // Validate the provided language against country's available languages
    const isValidLanguage =
        language &&
        countryConfig?.languages?.some(lang => lang.code === language);

    return {
        countryCode: countryConfig?.alpha2 || Alpha2.US,
        language: isValidLanguage ? language! : defaultLanguage
    };
}
