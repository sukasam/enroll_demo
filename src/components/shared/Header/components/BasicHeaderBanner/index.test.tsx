import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { Alpha2 } from "Constants/countryConfig/enums";
import { useFlags } from "launchdarkly-react-client-sdk";
import { useRouter } from "next/router";
import BasicHeaderBanner from "./index";

// Mock the dependencies
jest.mock("launchdarkly-react-client-sdk", () => ({
    useFlags: jest.fn()
}));

jest.mock("next/router", () => ({
    useRouter: jest.fn()
}));

jest.mock(
    "Components/shared/BannerNotificationAlerts/index",
    (): any =>
        function MockBannerNotificationAlerts({
            currentMarket,
            bannerFlagLD
        }: any) {
            return (
                <div
                    data-testid="banner-notification-alerts"
                    data-currentmarket={currentMarket}
                    data-bannerflagld={bannerFlagLD.toString()}
                />
            );
        }
);

describe("BasicHeaderBanner", () => {
    const mockProps = {
        currentMarket: Alpha2.US,
        flagNameLD: "testFlag"
    };

    beforeEach(() => {
        (useFlags as jest.Mock).mockReturnValue({ testFlag: true });
        (useRouter as jest.Mock).mockReturnValue({ pathname: "/test" });
    });

    it("renders BannerNotificationAlerts when flag is true", () => {
        render(<BasicHeaderBanner {...mockProps} />);
        expect(
            screen.getByTestId("banner-notification-alerts")
        ).toBeInTheDocument();
    });

    it('does not render on home page when flagNameLD is "outOfStock"', () => {
        (useFlags as jest.Mock).mockReturnValue({ outOfStock: true });
        (useRouter as jest.Mock).mockReturnValue({ pathname: "/home" });
        const { container } = render(
            <BasicHeaderBanner {...mockProps} flagNameLD="outOfStock" />
        );
        expect(container.firstChild).toBeNull();
    });

    it('renders on home page when flagNameLD is not "outOfStock"', () => {
        (useRouter as jest.Mock).mockReturnValue({ pathname: "/home" });
        render(<BasicHeaderBanner {...mockProps} />);
        expect(
            screen.getByTestId("banner-notification-alerts")
        ).toBeInTheDocument();
    });

    it("passes correct props to BannerNotificationAlerts", () => {
        render(<BasicHeaderBanner {...mockProps} />);
        const banner = screen.getByTestId("banner-notification-alerts");
        expect(banner).toHaveAttribute("data-currentmarket", Alpha2.US);
        expect(banner).toHaveAttribute("data-bannerflagld", "true");
    });

    it("matches snapshot when rendered", () => {
        const { container } = render(<BasicHeaderBanner {...mockProps} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
