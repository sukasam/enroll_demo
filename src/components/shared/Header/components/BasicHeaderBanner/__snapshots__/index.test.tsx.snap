// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicHeaderBanner matches snapshot when rendered 1`] = `
<div
  data-bannerflagld="true"
  data-currentmarket="US"
  data-testid="banner-notification-alerts"
/>
`;
